decoder.go

```diff
@@ -73,7 +73,7 @@ func convertSignedInts(src interface{}, dst *reflect.Value) error {
 			return nil
 		}
 	case uintptr:
-		if t <= math.MaxInt64 && !dst.OverflowInt(int64(t)) {
+		if int64(t) <= math.MaxInt64 && !dst.OverflowInt(int64(t)) {
```

Comment:
should we use `OverflowUint` for `uintptr`?

Reply:
This method is for setting `dst`, which is an integer in this method. `OverflowUint` will panic in this case.

Reply:
There is another bug in the code though,  int64(math.MaxUint64) will be -1 :)